tools/spm/partition_description_schema.json

```diff
@@ -0,0 +1,188 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "schema for a partition description.",
+  "type": "object",
+  "required": ["name", "type", "priority", "id", "entry_point", "stack_size", "heap_size", "source_files"],
+  "properties": {
+    "name": {
+      "description": "Alphanumeric C macro for referring to a partition. (all capital)",
+      "$ref": "#/definitions/c_macro"
+    },
+    "type": {
+      "description": "Whether the partition is unprivileged or part of the trusted computing base.",
+      "enum": ["SECURE", "TRUSTED"]
+    },
+    "priority": {
+      "description": "Partition task priority.",
+      "enum": ["LOW", "NORMAL", "HIGH"]
+    },
+    "id": {
+      "description": "Partition numeric unique positive identifier. (must be a positive 8 bytes hex string)",
+      "type": "string",
+      "pattern": "^0x[0-7][0-9a-fA-F]{7}$"
+    },
+    "entry_point": {
+      "description": "C symbol name of the partition's entry point. (unmangled, use extern C if needed)",
+      "$ref": "#/definitions/c_symbol"
+    },
+    "stack_size": {
+      "description": "Partition's task stack size in bytes.",
+      "$ref": "#/definitions/positive_integer_or_hex_string"
+    },
+    "heap_size": {
+      "description": "Partition's task heap size in bytes.",
+      "$ref": "#/definitions/positive_integer_or_hex_string"
+    },
+    "mmio_regions": {
+      "description": "List of Memory-Mapped IO region objects which the partition has access to.",
+      "type": "array",
+      "items": {
+        "anyOf": [{
+            "$ref": "#/definitions/named_region"
+          },
+          {
+            "$ref": "#/definitions/numbered_region"
+          }
+        ]
+      },
+      "uniqueItems": true
+    },
+    "secure_functions": {
+      "description": "List of Secure Function objects which the partition implements.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/secure_func"
+      },
+      "uniqueItems": true
+    },
+    "extern_sfids": {
+      "description": "List of SFID which the partition code depends on and allowed to access.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/c_macro"
+      },
+      "uniqueItems": true
+    },
+    "source_files": {
```

Comment:
This seems a little out of place in mbed OS. We seem to prefer scanning for source files, and using uppercase names to add semantics to paths. This field is list of files, which is _usually_ not found in Mbed OS. Perhaps there is a way to make this more like the Mbed OS build system? The first few ideas that I had were a worse user experience, so I'm not sure matching the Mbed OS build system is a great idea yet.

Reply:
you'll have to take this with the PSA architects

Reply:
@theotherjimmy PSA specification defines a way of describing Secure Partitions in the JSON manifest form. 
These secure partitions will eventually make their way in to TrustZone on ARMv8-M. 

The incentive for explicitly naming related source files is fine grained control of what sources are added in to TrustZone images.

Globbing for source files is very convenient for Non Secure image - but we should be extra careful when composing the secure image. The less code goes in translates to narrower attack surface.